File: Data Structures & Algorithms/maximum-product-subarray/submission-5.js
class Solution {
    /**
     * @param {number[]} nums
     * @return {number}
     */
    maxProduct(nums) {
        let n = nums.length;
        let res = nums[0];
        let prefix = 0; 
        let suffix = 0;

        for(let i=0;i <n;i++){
            prefix = nums[i] * (prefix === 0 ? 1 : prefix);
            suffix = nums[n - 1 - i] * (suffix === 0 ? 1 : suffix);
            res = Math.max(res, Math.max(prefix, suffix))
        }
        return res === -0 ? 0 : res;
    }
}

File: Data Structures & Algorithms/longest-consecutive-sequence/submission-5.java
class Solution {
    public int longestConsecutive(int[] nums) {
        HashSet<Integer> hs = new HashSet<>();
        for(int num: nums){
            hs.add(num);
        }
        int max = 0;
        for(int num: hs) {
            if(!hs.contains(num -1)) {
                int count = 1;
                int curr = num;

                while(hs.contains(curr+1)) {
                    curr++;
                    count++;
                }
                max = Math.max(max, count);
            }
        }
        return max;
    }
}
